vendor/riscv/riscv-isa-sim/riscv/csrs.cc

```diff
@@ -992,13 +993,15 @@ bool wide_counter_csr_t::unlogged_write(const reg_t val) noexcept {
   // takes precedence over the increment to instret.  However, Spike
   // unconditionally increments instret after executing an instruction.
   // Correct for this artifact by decrementing instret here.
-   openhw::reg::unlogged_write( openhw::reg::unlogged_read()-1);
+
+  // disable since we (at the moment) check mcounterinhibit
+  //openhw::reg::unlogged_write( openhw::reg::unlogged_read()-1);
   return true;
 }
 
 reg_t wide_counter_csr_t::written_value() const noexcept {
   // Re-adjust for upcoming bump()
-  return  openhw::reg::unlogged_read() + 1;
+  return  openhw::reg::unlogged_read(); //+ 1; // Disable since we dont always bump
```

Comment:
The +1 caused errors in `cv32e40s_csr_access_test` which reads the `mcycle` csr. Spike did not support disabling and enabling counting of `mcycle` with `mcountinhibit`. In 40S counting is disabled by default, so this caused a mismatch when the `mcycle` csr was read. Here `mcycle` counted up in Spike but not in the core, since `mcountinhibit` was enabled (disabling counting).